### PR TITLE
Bonus Script clean up

### PR DESCRIPTION
**Pull request by cydh**
* Removed fixed array size (MAX_PC_BONUS_SCRIPT) of bonus_script on players.
* Changed fixed script string on player's bonus_script to StringBuf, reduce memory usage for each connected player.
* Added new flags to expand duplicated script and flag to allow duplicate script.
* Reduced buf lenght from 1024 to 512 on bonus_script_data (used for saving/loading bonus_script from char-server). Still, MAX_PC_BONUS_SCRIPT is used to gives limit data can be loaded.

NOTE:
Please import 'upgrade_20150131.sql' for `bonus_script` table changes

Signed-off-by: Cydh Ramdh <house.bad@gmail.com>

### CODE REVIEW

**Comment by cydh** [comment]
still, there's something wrong when removing bonus_script (timed out or removed).
I want change it to link list.

**Comment by icxbb-xx** [comment]
I have test , but found something wrong about timer status show wrong in client

ex.
14602,TaogunkaS,Tao Gunka Scroll,2,0,,10,,,,,0xFFFFFFFF,63,2,,,,,,{ specialeffect2 EF_LIGHTSPHERE; bonus_script "{ bonus bMaxHPrate,100; bonus bDefRate,-50; bonus bMdefRate,-50; }",180,0,0,SI_MVPCARD_TAOGUNKA; },{},{}

Show 5 hr

**Comment by cydh** [comment]
@icxbb-xx fixed on newer commit

File: src/map/pc.c [hunk 1]
```diff
@@ -11038,80 +11039,219 @@ void pc_show_version(struct map_session_data *sd) {
...
 }
 
-/** [Cydh]
+int pc_bonus_script_list(void *key, void *data, va_list ap) {
+	struct s_bonus_script_entry *entry = (struct s_bonus_script_entry *)data;
+	struct map_session_data *sd = va_arg(ap, struct map_session_data *);
+	struct linkdb_node *node = (struct linkdb_node *)key;
+	if (sd)
+		ShowDebug("  cid=%d aid=%d\n",sd->status.char_id, sd->status.account_id);
+	ShowDebug("    key:%d e:0x%08X n:0x%08X nn:0x%08X np:0x%08X\n",(intptr_t)key, entry, key, node->next, node->prev);
```
Comment (cydh) [inline src/map/pc.c]: D: debug msg leftovers xD

File: src/map/chrif.c [hunk 2]
```diff
@@ -1954,6 +1965,13 @@ void do_init_chrif(void) {
 			sizeof(struct mmo_charstatus));
 		exit(EXIT_FAILURE);
 	}
+
+	if((sizeof(struct bonus_script_data) * MAX_PC_BONUS_SCRIPT) > 0xFFFF){
+		ShowError("bonus_script_data size = %d is too, please reduce MAX_PC_BONUS_SCRIPT (%d) size. (must be below 0xFFFF).\n",
```
Comment (cydh) [inline src/map/chrif.c]: _too_ what? -a

**Comment by cydh** [comment]
clicked :v

**Pull request merged**